src/Cli/dotnet/Commands/Run/RunProperties.cs

```diff
@@ -19,17 +19,22 @@ internal RunProperties(string command, string? arguments, string? workingDirecto
     {
     }
 
-    internal static RunProperties FromProject(ProjectInstance project)
+    internal static bool TryFromProject(ProjectInstance project, out RunProperties result)
     {
-        var result = new RunProperties(
+        result = new RunProperties(
             Command: project.GetPropertyValue("RunCommand"),
             Arguments: project.GetPropertyValue("RunArguments"),
             WorkingDirectory: project.GetPropertyValue("RunWorkingDirectory"),
             RuntimeIdentifier: project.GetPropertyValue("RuntimeIdentifier"),
             DefaultAppHostRuntimeIdentifier: project.GetPropertyValue("DefaultAppHostRuntimeIdentifier"),
             TargetFrameworkVersion: project.GetPropertyValue("TargetFrameworkVersion"));
 
-        if (string.IsNullOrEmpty(result.Command))
+        return !string.IsNullOrEmpty(result.Command);
```

Comment:
It seems a little strange for `result` to contain non-null when the return value is `false`.

Reply:
Yeah, since we already allocated the RunProperties, I returned them, but it might be an anti-pattern, I will fix, thanks.